pkg/snapstore/abs_snapstore.go

```diff
@@ -78,69 +83,62 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 }
 
 // ConstructBlobServiceURL constructs the Blob Service URL based on the activation status of the Azurite Emulator.
-// It checks the environment variables for emulator configuration and constructs the URL accordingly.
-// The function expects two environment variables:
+// It checks the environment variable for emulator configuration and constructs the URL accordingly.
+// The function expects the following environment variable:
 // - AZURE_EMULATOR_ENABLED: Indicates whether the Azurite Emulator is enabled (expects "true" or "false").
-// - AZURE_STORAGE_API_ENDPOINT: Specifies the Azurite Emulator endpoint when the emulator is enabled.
-func ConstructBlobServiceURL(credentials *azblob.SharedKeyCredential) (*url.URL, error) {
-	defaultURL, err := url.Parse(fmt.Sprintf("https://%s.%s", credentials.AccountName(), brtypes.AzureBlobStorageHostName))
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse default service URL: %w", err)
-	}
+func ConstructBlobServiceURL(storageAccount, domain string) (*url.URL, error) {
+	scheme := "https"
+
 	emulatorEnabled, ok := os.LookupEnv(EnvAzureEmulatorEnabled)
-	if !ok {
-		return defaultURL, nil
-	}
-	isEmulator, err := strconv.ParseBool(emulatorEnabled)
-	if err != nil {
-		return nil, fmt.Errorf("invalid value for %s: %s, error: %w", EnvAzureEmulatorEnabled, emulatorEnabled, err)
-	}
-	if !isEmulator {
-		return defaultURL, nil
-	}
-	endpoint, ok := os.LookupEnv(AzuriteEndpoint)
-	if !ok {
-		return nil, fmt.Errorf("%s environment variable not set while %s is true", AzuriteEndpoint, EnvAzureEmulatorEnabled)
+	if ok {
+		isEmulator, err := strconv.ParseBool(emulatorEnabled)
+		if err != nil {
```

Comment:
Because right now, we still don't support Azurite HTTPS. So until then, we will need this, simply to set `http` scheme for the URL.

Reply:
We could just break compatibility with Azurite for a few commits, and get rid of this altogether. What are your opinions?

Reply:
We can remove the env var when we have consistency between the URLs for Azure real infra and Azurite. Unless someone picks it up soon and does it, I would not want to remove this nice functionality of being able to test locally using Azurite.

Reply:
@renormalize @unmarshall on second thought, I realise that without support for taking the full endpoint as the input, it's not possible to set the Azurite endpoint for the snapstore here, since this constructs the URL as `http(s)://<storageAccount>.<domain>`, which does not match the format for the Azurite URL `http://<endpoint>` (maybe @renormalize you have a better idea about this). So it looks like we might need to break Azurite support for a few commits anyway.